### PR TITLE
[Text Generation][Enhancement] If `prompt_processing_sequence_length` == 1, do not inititalize multitoken_engine

### PR DESCRIPTION
**Pull request by dbogunowicz**
A simple improvement that streamlines the pipeline.

If `prompt_processing_sequence_length == 1`, we are essentially running single-token prompt prefill , so we should not be initializing and running the additional engine.

### CODE REVIEW

File: src/deepsparse/transformers/pipelines/text_generation.py [hunk 1]
```diff
@@ -680,20 +685,14 @@ def engine_inputs_for_prefill(
...
-                            .reshape(1, -1)
-                            .astype(numpy.int64)
+                    engine_input = (
+                        numpy.arange(
+                            num_cached_entries,
+                            num_cached_entries + self.prompt_processing_sequence_length,
                         )
+                        .reshape(1, -1)
+                        .astype(numpy.int64)
+                    )
```
Comment (tlrmchlsmth) [inline src/deepsparse/transformers/pipelines/text_generation.py:695]: what's the reason for this change?

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:695]: Handling the case where `prompt_processing_sequence_lenght == 1` vs `!=1` required some additional lines of code. If you look closely, I am actually removing these superfluous lines of code, and keeping only the essential logic for the creation of the engine_inputs for multitoken_engine.

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:695]: (lines 684-686 are not needed anymore)

Comment (tlrmchlsmth) [reply src/deepsparse/transformers/pipelines/text_generation.py:695]: Will this function no longer work when `self.prompt_processing_sequence_length == 1`? Should we add an assert?

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:695]: Those lines will be never triggered, will never end up using the multitoken engine in this scenario.

**Review by mgoin — APPROVED** [review]
awesome! a special case i'll take advantage of :)
